### ISSUE TITLE
Overriding methods from Abstract classes in the subclass, and using the overriden methods back in the Abstract super class

### ISSUE THREAD

**Issue by Keithlimhs** [nus-cs2030/2021-s2#42]
# Summary

I have created an Abstract class Service which serves as a super class for the 3 services offered, and abstract methods which gets the Fare of the subclass that inherits Service. I would like to know if it is possible to override the abstract methods of the superclass Service in the respective subclasses and then using the overriden methods back in the superclass toString() method, so as to minimize code repetition.

## Description:

Clarifications or details with respect to the issue you are facing

## Code Snippets

Attach code snippets (if any)

## Screenshots

Screenshots (if any)

**Comment by raymondtanujaya** [comment]
Yes. In fact, when you decided that the method(e.g. computeFare) is abstract, you are asking the subclass to override it since it has not been implemented(abstract). Then, since your Service actually knows and does have some abstract methods earlier(yet to be implemented by subclass), it can call them in toString or other methods.

**Comment by OuaisBien** [comment]
do you mean you want to use the toString() method in the superclass after it is overriden?

**Comment by Keithlimhs** [comment]
> do you mean you want to use the toString() method in the superclass after it is overriden?

The toString() method in the superclass is overriden and calls the abstract method that is overriden in the subclass. The subclasses does not have overriden toString() method. Was wondering if that is possible

**Comment by Keithlimhs** [comment]
> Yes. In fact, when you decided that the method(e.g. computeFare) is abstract, you are asking the subclass to override it since it has not been implemented(abstract). Then, since your Service actually knows and does have some abstract methods earlier(yet to be implemented by subclass), it can call them in toString or other methods.
> 
> Furthermore, your toString is an example of a good use of abstract class rather than interface in the sense that you can gather that commonality and implement it using a **concrete** method( compare it to getVolume in shape3D where you can't exactly define a one-for-all, **concrete** method). CMIIW

Oh I see thank you so much for the clarification
